src/NUnitEngine/nunit.engine/Runners/DirectTestRunner.cs

```diff
@@ -86,10 +98,19 @@ protected override TestEngineResult LoadPackage()
             foreach (var subPackage in packages)
             {
                 var testFile = subPackage.FullName;
+
+                if (_assemblyResolver != null && !TestDomain.IsDefaultAppDomain()
+                    && subPackage.GetSetting(PackageSettings.ImageRequiresDefaultAppDomainAssemblyResolver, false))
+                {
+                    _assemblyResolver.AddPathFromFile(testFile);
+                }
+
                 IFrameworkDriver driver = driverService.GetDriver(TestDomain, testFile);
                 driver.ID = TestPackage.ID;
                 result.Add(driver.Load(testFile, subPackage.Settings));
                 _drivers.Add(driver);
+
+                _packageIndex.Add(driver, subPackage);
```

Comment:
I should clarify on my previous question though, simply having each user track which entries they added and then remove those would be in error if it's possible for multiple users to add the same path. The shared resolver would need to count how many adds and removes have been issues for each path. Minor complication.